patches/all/@oclif+dev-cli+1.26.0.patch

```diff
@@ -0,0 +1,250 @@
+     const packCLI = async () => {
+         const stdout = await qq.x.stdout('npm', ['pack', '--unsafe-perm'], { cwd: c.root });
+         return path.join(c.root, stdout.split('\n').pop());
+@@ -30,11 +31,19 @@ async function build(c, options = {}) {
+         tarball = path.basename(tarball);
+         tarball = qq.join([c.workspace(), tarball]);
+         qq.cd(c.workspace());
+-        await qq.x(`tar -xzf ${tarball}`);
++        const { msysExec, toMsysPath } = require("../util");
++        await msysExec(`tar -xzf ${toMsysPath(tarball)}`);
+         // eslint-disable-next-line no-await-in-loop
+         for (const f of await qq.ls('package', { fullpath: true }))
+             await qq.mv(f, '.');
+         await qq.rm('package', tarball, 'bin/run.cmd');
++        // rename the original balena-cli ./bin/balena entry point for oclif compatibility
++        await qq.mv('bin/balena', 'bin/run');
++        // The oclif installers are a production installation, while the source
++        // `bin` folder may contain a `.fast-boot.json` file of a dev installation.
++        // This has previously led to issues preventing the CLI from starting, so
++        // delete `.fast-boot.json` (if any) from the destination folder.
++        await qq.rm('bin/.fast-boot.json');
+     };
+     const updatePJSON = async () => {
+         qq.cd(c.workspace());
+@@ -46,21 +55,21 @@ async function build(c, options = {}) {
+         await qq.writeJSON('package.json', pjson);
+     };
+     const addDependencies = async () => {
+-        qq.cd(c.workspace());
+-        const yarnRoot = findYarnWorkspaceRoot(c.root) || c.root;
+-        const yarn = await qq.exists([yarnRoot, 'yarn.lock']);
+-        if (yarn) {
+-            await qq.cp([yarnRoot, 'yarn.lock'], '.');
+-            await qq.x('yarn --no-progress --production --non-interactive');
+-        }
+-        else {
+-            let lockpath = qq.join(c.root, 'package-lock.json');
+-            if (!await qq.exists(lockpath)) {
+-                lockpath = qq.join(c.root, 'npm-shrinkwrap.json');
+-            }
+-            await qq.cp(lockpath, '.');
+-            await qq.x('npm install --production');
++        const ws = c.workspace();
++        qq.cd(ws);
++        console.error(`[debug] @oclif/dev-cli copying node_modules to "${ws}"`)
++        const source = path.join(c.root, 'node_modules');
++        if (process.platform === 'win32') {
++            // xcopy is much faster than `qq.cp(source, ws)`
++            await qq.x(`xcopy "${source}" "${ws}\\node_modules" /S /E /B /I /K /Q /Y`);
++        } else {
++            // use the shell's `cp` on macOS in order to preserve extended
++            // file attributes containing `codesign` digital signatures
++            await qq.x(`cp -pR "${source}" "${ws}"`);
+         }
++        console.error(`[debug] @oclif/dev-cli running "npm prune --production" in "${ws}"`);
++        await qq.x('npm prune --production');
```

Comment:
This is the main change that helps with the notarization of macOS installers. Previously (code deleted 15 lines above), oclif would run `npm install --production` to produce a production install of the CLI that would then be packaged in the GUI installer. With these changes, we instead clone the `node_modules` folder of a development installation, preserving extended file attributes such as the macOS `codesign` digital signature of `.node` native module files (PR #2186), and then run `npm prune --production` to remove dev packages. This is similar to what we've already been doing for the CLI Docker images (check [this line](https://github.com/balena-io/balena-cli/blob/v12.42.2/docker/debian/Dockerfile#L17)), so it's been already tested a bit, and I found that it is also overall faster than running `npm install --production`, as it avoids the compilation of the native node modules.